### PR TITLE
Hulks can now swing people by their tails + small changes to carbon throw impacts

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
[Official PR theme](https://www.youtube.com/watch?v=e3VUjIev1-0)

This PR adds another little interaction to hulks, letting them swing monkeys and humanoids with tails (so lizardpeople and felinids mainly) similar to (copied from) the wrestling move. To do this, be a hulk, have a carbon with a tail in at least a neck grab, target the groin, enable throw mode, then click in the direction you'd like to chuck them. After a short 2 second do_after, you'll start swinging your victim around and around by the tail, eventually sending them flying at a very, very high speed! The full process takes about 14 seconds from start to finish, not including getting them in a neck grab to begin with. Impacting a solid wall with no armor will do about 30 brute damage and can cause up to a hairline fracture if you're very unlucky.

[Here's a clip of it in action](https://cdn.discordapp.com/attachments/585862469508005888/785432666383974420/so_long.mp4)

This PR also makes a few changes to the effects of being thrown into things. Previously, you'd only suffer damage and a 2 second paralyze if you were thrown into solid turfs or carbon mobs, which meant smashing into windows or girders at high speed didn't do anything. This PR makes it so high speed impacts (I.E. hulk throws and high speed disposal outputs, not normal body throwing) into solid atoms will still deal a lesser amount of damage. Structures like windows and girders will deal a bit more damage and a small 1 second paralyze compared to other atoms

## Why It's Good For The Game
Tail spinning adds a goofy niche power to hulk that doesn't just involve deleting walls and shrugging off stuns, and the other throw impact changes make being thrown into things at high speed more consistent rather than only applying to a specific type of atom.

## Changelog
:cl: Ryll/Shaps
add: Hulks can now throw monkeys/people with tails around by their tails! To do this, get your victim in at least a neck grab, enable throw mode, then click in the direction you want to throw. So long!
tweak: Carbons will now take damage when thrown at high speeds into things like windows, girders, and other solid objects, in addition to just walls and other people.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
when will hulks be able to vendor throw people?

**Comment by ATH1909** [comment]
I think letting this throw work off of an aggressive grab would be fine- the move itself has a 2 second do_after wind-up anyway.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -81,23 +81,28 @@
...
+	if(istype(throwingdatum))
 		hurt = !throwingdatum.gentle
-	if(hit_atom.density && isturf(hit_atom))
-		if(hurt)
-			Paralyze(20)
+	if(hurt && hit_atom.density)
+		if(isturf(hit_atom))
+			Paralyze(2 SECONDS)
 			take_bodypart_damage(10 + 5 * extra_speed, check_armor = TRUE, wound_bonus = extra_speed * 5)
+		else if(isstructure(hit_atom) && extra_speed)
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/carbon.dm:92]: does this affect throwing people into vending machines?

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -81,23 +81,28 @@
...
+			Paralyze(1 SECONDS)
+			take_bodypart_damage(5 + 5 * extra_speed, check_armor = TRUE, wound_bonus = extra_speed * 5)
+		else if(!iscarbon(hit_atom) && extra_speed)
+			take_bodypart_damage(5 * extra_speed, check_armor = TRUE, wound_bonus = extra_speed * 5)
 	if(iscarbon(hit_atom) && hit_atom != src)
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/carbon.dm:97]: note that this isn't an elseif()- make sure that throwing someone into a dense carbon at a higher than normal speed won't damage them twice.

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/mob/living/carbon/carbon.dm:92]: also, why do the structure and not-carbon checks check for extra_speed, but not the turf check? shouldn't the former be changed to match the standards of the latter?

**Comment by Krysonism** [comment]
It should do way more damage, 14 seconds build up to deal 30 damage is sad.

**Comment by Dorsisdwarf** [comment]
So long, gay bowser.

**Comment by jacklecroy** [comment]
This is just from the wrestling belt, you should modularize that.

**Comment by ilikesaying** [comment]
So Long, Gay Felinid

**Comment by optimumtact** [comment]
> It should do way more damage, 14 seconds build up to deal 30 damage is sad.

state of powergamers

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/mob/living/carbon/carbon.dm:92]: Walls and turfs are harder surfaces, so any impact with them is bad, whereas windows you can bounce off. Also vending machines check tipping in their hitby() so they won't be affected

**Comment by Acer202** [comment]
Does this still work if the tail is covered by a harduit and isn't visible?

**Comment by Ryll-Ryll** [comment]
> Does this still work if the tail is covered by a hardsuit and isn't visible?

Good catch, any suits that hide the tail will now protect against tailswings

**Pull request merged**